### PR TITLE
Add `alt-text` & remove `<figcaption>` as discussed in #65

### PR DESCRIPTION
**Pull request by dance2die**
# 🤔 What?

Added `alt-text` & remove `<figcaption>` as discussed in #65.

# ➕ Additional Context

I've created one giant PR as it'd be easier to skim thru the changes without having to go thru each commit.

And I've removed `<figcaption>` where
1. it's used as a simple `alt-text` replacement
1. it blocks rendering hyperlinks (refer to https://kentcdodds.com/blog/migrating-to-jest).

links under figcaption not rendered

I've left `<figcaption>` as they are when they are used as a description by itself or as a detail or an image/sandbox

### CODE REVIEW

File: content/blog/reacts-new-context-api/index.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ bannerCredit:
 Have you heard of the context API in React? If you've heard of it, are you like
 many others afraid to use it directly because you saw this in the official docs:
 
-![](./images/0.png)
+![search result of "context" on the official react documentation site](./images/0.png)
```
Comment (kentcdodds) [inline content/blog/reacts-new-context-api/index.md]: These can't have `"` in them (I think that's an MDX bug). We can just remove the `"` from all of these I think.

**Review by kentcdodds — CHANGES_REQUESTED** [review]
WOW! This is super! Thank you! Just one thing.

File: content/blog/reacts-new-context-api/index.md (hunk 1 above)
Comment (dance2die) [reply content/blog/reacts-new-context-api/index.md]: Thank you for the review & let me learn of such an issue 🙂~ 
I will go through such cases and update.

**Comment by dance2die** [comment]
I've fixed all `alt-text`s matching the following `regex` pattern
```regex
!\[.*(".*"?)\]
```

regex pattern in vscode

**Review by kentcdodds — APPROVED** [review]
Thank you a bunch!

**Comment by dance2die** [comment]
Thank you, Kent.
It's the least I can do for how you've helped me & others 🙂🙌

**Pull request merged**